### DISCUSSION TITLE
Fastlane concurrent/parallel builds with uniq build number

### DISCUSSION THREAD

**Discussion by sgabenov** [fastlane/fastlane#21054] [Q&A]
Hi all, i want to ask the community how do you coupe with problem with setting up uniq build version for builds that may run in parallel? Fastlane have the tool to set up build version based on the latest build version that is taken from the store and rised +1. But what if i have started 2 builds in parallel or in very short interval, and both versions have taken from the store the same version? It this case only first build will be able to be upladed to store, while the second one will fail, as it will have the same latest_version + 1 build version. It seems to have some additional service running in parallel that will chech the latest version from the store and will provide increment for each build (for instance reddis or something similar). Can you give your ideas how to coupe with such problem? The best way not to use any additional service in parallel. And we could not disable concurent builds for the app :(

**Comment by mdgeist** [comment]
Sounds like you need some CI job setup in whatever tools you'd like that increments a build number each time the job is ran. Normally its pretty basic and if you ran say the same job twice each run would have a unique build number.

**Comment by HenryGrange** [comment]
> Hi all, i want to ask the community how do you coupe with problem with setting up uniq build version for builds that may run in parallel? Fastlane have the tool to set up build version based on the latest build version that is taken from the store and rised +1. But what if i have started 2 builds in parallel or in very short interval, and both versions have taken from the store the same version? It this case only first build will be able to be upladed to store, while the second one will fail, as it will have the same latest_version + 1 build version. It seems to have some additional service running in parallel that will chech the latest version from the store and will provide increment for each build (for instance reddis or something similar). Can you give your ideas how to coupe with such problem? The best way not to use any additional service in parallel. And we could not disable concurent builds for the app :(

Hi I have the same problem did you find a solution? Thanks
